gabbi/case.py

```diff
@@ -419,7 +437,8 @@ def _test_data_to_string(self, data, content_type):
                 else:
                     return info
         else:
-            data = json.dumps(data)
+            data = self.replace_template(data, retype_numerals=True)
```

Comment:
The reason for the reordering here is so that when data is passed to replace_template it is a dict or list instead of string. Since replace_template is recursive, each chunk of data will be processed in a small way, so that only chunks which are just a number will become numbers.

However you've just caused me to find a possible bug, or a total flaw in the logic, so there will be at least one more commit.